rag/rag-base/src/commonMain/kotlin/ai/koog/rag/base/files/FileSystemProvider.kt

```diff
@@ -4,240 +4,241 @@ import kotlinx.io.Sink
+         * Computes the relative path from a [root] to a target [path].
+         * It doesn't check if the paths actually exist in the filesystem.
          *
-         * @param root The base path against which the relative path should be computed.
-         * @param path The target path to compute its relation to the root.
-         * @return A string representing the relative path from the root to the path, or null if the paths cannot be relativized.
+         * @param root The root path.
+         * @param path The target path.
+         * @return The relative path as a string, or null if the paths cannot be relativized (e.g., they have no common prefix).
          */
-        public suspend fun relativize(root: Path, path: Path): String?
+        public fun relativize(root: Path, path: Path): String?
+
         /**
-         * Checks if a given path exists in the file system.
+         * Checks if a [path] exists in the filesystem.
          *
-         * @param path The path to be checked for existence.
-         * @return `true` if the path exists, `false` otherwise.
+         * @param path The path to check.
+         * @return true if the path exists, false otherwise.
          */
         public suspend fun exists(path: Path): Boolean
     }
 
     /**
-     * An interface that defines the operations for reading file or content-related data.
-     * Extends the [Serialization] interface to provide additional methods for reading byte data,
-     * retrieving data sources, and determining the size of given paths.
+     * Provides operations for reading file content.
      *
-     * @param Path The type that represents the file or directory path.
+     * @param Path The type representing file paths in the implementation.
      */
     public interface Read<Path> : Serialization<Path> {
         /**
-         * Reads the content of the file at the specified path.
+         * Reads the content of a file at the specified [path].
          *
-         * @param path The path of the file to read.
-         * @return The content of the file as a ByteArray.
+         * @param path The path to read.
+         * @return The file content as a byte array.
+         * @throws IllegalArgumentException if the path doesn't exist or isn't a regular file.
          */
         public suspend fun read(path: Path): ByteArray
+
         /**
-         * Provides access to a data source represented by the given path.
+         * Creates a Source for reading from a file at the specified [path].
+         * The returned Source is buffered.
          *
-         * @param path The path to the source from which data should be accessed.
-         * @return A Source object associated with the specified path.
+         * @param path The path to read from.
+         * @return A buffered Source object for reading.
+         * @throws NoSuchFileException if the path doesn't exist.
+         * @throws IllegalArgumentException if the path isn't a regular file.
```

Comment:
I'd merge these two cases into one `IllegalArgumentException` to make it similar to `read`